packages/playwright-core/src/server/chromium/crNetworkManager.ts

```diff
@@ -367,7 +367,10 @@ export class CRNetworkManager {
     // For frame-level Requests that are handled by a Service Worker's fetch handler, we'll never get a requestPaused event, so we need to
     // manually create the request. In an ideal world, crNetworkManager would be able to know this on Network.requestWillBeSent, but there
     // is not enough metadata there.
-    if (!request && event.response.fromServiceWorker) {
+    //
+    // PW_EXPERIMENTAL_SERVICE_WORKER_NETWORK_EVENTS we guard with, since this would fix an old bug where, when using routing,
+    // request would not be emitted to the user for requests made by a page with a SW (and fetch handler) registered
+    if (!!process.env.PW_EXPERIMENTAL_SERVICE_WORKER_NETWORK_EVENTS && !request && event.response.fromServiceWorker) {
```

Comment:
This is debatable. On 1.23 (and presumably all versions prior), https://github.com/microsoft/playwright/pull/15688/files#diff-f48a7787820cb49396256c1234745152eb05ca5a3f43b6d128ce0ee45eb6dff6R609 would timeout. If I don't flag this new logic off, it doesn't timeout.

We can debate preserving the old behavior or not.

Reply:
The only concern in fixing the old behavior, is having "extra" events show up for users that already have SWs on their pages, and are waiting for events like `page.on('request')`